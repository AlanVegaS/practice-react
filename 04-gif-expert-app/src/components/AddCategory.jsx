import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')
    const onInputChange = () => {
        setInputValue(event.target.value)
    }
    const onSubmit = () => {
        event.preventDefault()
        const value = inputValue.trim()
        if(!value) return
        //setCategories([...categories, inputValue])
        onNewCategory(value)
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Find gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}