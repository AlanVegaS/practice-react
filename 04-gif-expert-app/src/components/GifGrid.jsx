import { GifCard } from "./GifCard";
import { useFetchGifs } from "./hooks/useFetchGifs";

export const GifGrid = ({category}) => {
    const {images, isLoading} = useFetchGifs(category)

    return (
        <>
            <h2>{category}</h2>
            {
                isLoading && (<h3>Loading...</h3>)
            }
            <div className="img-section">
            {
                images.map((gif) => (
                    <GifCard 
                        key = {gif.id}
                        {...gif}
                    />
                ))
            }
            </div>
        </>
    )
}