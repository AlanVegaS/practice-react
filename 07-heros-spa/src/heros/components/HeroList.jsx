import React, { useMemo } from "react";
import { getHerosBtPublisher } from "../helpers";
import { HeroCard } from "./HeroCard";

export const HeroList = ({ publisher }) => {

    const heros = useMemo(() => getHerosBtPublisher(publisher), [publisher])

    return (
        < div className="row rows-cols-1 row-cols-md-3 g-3">
            {
                heros.map(hero => (
                    <HeroCard key={hero.id} {...hero}/>
                ))
            }
        </div >
    )
}