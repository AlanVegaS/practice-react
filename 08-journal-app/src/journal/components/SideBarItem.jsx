import { TurnedInNot } from '@mui/icons-material'
import { Grid, ListItem, ListItemButton, ListItemIcon, ListItemText } from '@mui/material'
import { useMemo } from 'react';
import { useDispatch } from 'react-redux';
import { setActivateNote } from '../../store/journal';

export const SideBarItem = ({ title, body, id, date, imageUrls }) => {

    const newTitlee = useMemo(() => {
        return title.length > 17
            ? title.substring(0, 17) + '...'
            : title;
    }, [title]);

    const dispatch = useDispatch();

    const onClickNote = (event) => {
        dispatch(setActivateNote({ title, body, id, date, imageUrls }))
    }

    return (
        <ListItem disablePadding>
            <ListItemButton onClick={onClickNote}>
                <ListItemIcon>
                    <TurnedInNot />
                </ListItemIcon>
                <Grid container>
                    <ListItemText primary={newTitlee} />
                    <ListItemText secondary={body} />
                </Grid>
            </ListItemButton>
        </ListItem>
    )
}
