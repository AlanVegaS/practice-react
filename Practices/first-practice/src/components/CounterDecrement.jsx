import { useForm } from "../hooks/useForm";

export const CounterDecrement = () => {

    const { OnChangeInput, setNewValueCounter, count } = useForm(1);
    const newInputValue = (evnt) => {
        const { value } = evnt.target;
        setNewValueCounter(value * 1);

    }

    return (<>
        <h3>Current count is: {count}</h3>
        <button onClick={() => setNewValueCounter(count - 1)}>-</button >
        <button onClick={() => setNewValueCounter(count + 1)}>+</button>
        <div>
            <input type="text" onChange={newInputValue} value={count}/>
        </div>
    </>);
}