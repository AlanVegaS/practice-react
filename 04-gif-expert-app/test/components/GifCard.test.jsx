import { render, screen } from '@testing-library/react'
import { GifCard } from "../../src/components/GifCard";

describe('Pruebas en GifCard', () => {
    const title = 'Hola'
    const url = 'http://localhost/url'
    test('Debe hacer match con el snap', () => {
        const { container } = render(<GifCard title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('Debe tner alt y url', () => {
        render(<GifCard title={title} url={url} />)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifCard title={title} url={url} />)
        expect(screen.getByAltText(title)).toBeTruthy()
    })
}); 