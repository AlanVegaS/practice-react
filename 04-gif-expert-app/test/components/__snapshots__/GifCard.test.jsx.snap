// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifCard Debe hacer match con el snap 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
      src="http://localhost/url"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
